src/woo_publications/api/tests/test_pagination.py

```diff
@@ -0,0 +1,113 @@
+from django.urls import path
+
+from rest_framework import generics, serializers, status
+from rest_framework.test import APITestCase, URLPatternsTestCase
+
+from woo_publications.metadata.models import Organisation
+from woo_publications.metadata.tests.factories import OrganisationFactory
+
+from ..pagination import DynamicPageSizePagination
+
+EXPECTED_PAGE_SIZE = 10
+EXPECTED_MAX_PAGE_SIZE = 100
+
+
+class TestDynamicPageSizePaginationSerializer(serializers.ModelSerializer):
+    class Meta:  # pyright: ignore
+        model = Organisation
+        fields = ("uuid",)
+
+
+class TestDynamicPageSizePaginationViewSet(generics.ListAPIView):
+    queryset = Organisation.objects.all()
+    serializer_class = TestDynamicPageSizePaginationSerializer
+    pagination_class = DynamicPageSizePagination
+    authentication_classes = ()
+    permission_classes = ()
+
+    """
+    A simple ViewSet for listing or retrieving users.
+    """
+
+
+class DynamicPageSizePaginationTest(URLPatternsTestCase, APITestCase):
+    urlpatterns = [
+        path("pagination", TestDynamicPageSizePaginationViewSet.as_view()),
+    ]
+
+    def test_get_default_pagination(self):
+        OrganisationFactory.create_batch(10)
+        response = self.client.get("/pagination")
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        data = response.json()
+
+        self.assertEqual(data["count"], 10)
+        self.assertEqual(len(data["results"]), EXPECTED_PAGE_SIZE)
+        self.assertIsNone(data["next"])
+        self.assertIsNone(data["previous"])
+
+    def test_alter_page_size(self):
+        OrganisationFactory.create_batch(15)
+
+        with self.subTest("no page defined"):
+            response = self.client.get("/pagination?pageSize=5")
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+            data = response.json()
+
+            self.assertEqual(data["count"], 15)
+            self.assertEqual(len(data["results"]), 5)
+            self.assertTrue(data["next"])
+            self.assertIsNone(data["previous"])
+
+        with self.subTest("page 1"):
+            response = self.client.get("/pagination?pageSize=5&page=1")
```

Comment:
please don't manually build querystrings, instead use the dict argument for the params.